src/metabase/query_processor/pivot.clj

```diff
@@ -357,13 +357,14 @@
         metadata-provider  (or (:lib/metadata query)
                                (lib.metadata.jvm/application-database-metadata-provider (:database query)))
         query              (lib/query metadata-provider query)
-        index-in-breakouts (into {}
-                                 (comp (filter (comp #{:source/breakouts} :lib/source))
-                                       (map-indexed (fn [i column] [(:name column) i])))
-                                 (lib/returned-columns query))
+        returned-columns   (lib/returned-columns query)
+        column-name->index (into {}
+                                 (map-indexed (fn [i column] [(:name column) i]))
+                                 (concat (filter (comp #{:source/breakouts} :lib/source) returned-columns)
```

Comment:
we need breakouts before aggregations (see https://github.com/metabase/metabase/blob/79c1d763c20edbb4a59318b6d105ac386fe972c7/src/metabase/query_processor/pivot.clj#L393). If there is a cleaner way please let me know